### PR TITLE
Make `border=none` setting work in tables

### LINKED ISSUE #6841: How to make table border=none setting work?
## 📝 Provide a description of the improvement

If you set table borders to none this is what you'll likely see:

The table still has a border. Why does it happen? It's because the table is styled by [default content styles](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/content-styles.html#the-full-list-of-content-styles) to this:

These styles are applied by default **via a stylesheet** to every table in the editor. Since the "none" style set in the "Table properties" balloon does **not** apply any inline style to the `<table>` element, the styles coming from the stylesheet are applied.

Therefore, in order to make a table not have any border by default one needs to add such a CSS to their website's styles (note: `td`s need to be styled too):

```css
body .ck-content .table table, body .ck-content .table td {
	border: none;
}
```

With this style, you'll get:

However, this isn't perfect – the table is now rendered in a very unclear way. Therefore, it's better to load different stylesheet in a page where the editor is rendered and in a page where the editor contents is rendered:

```css
/* page-with-editor.css */
body .ck-content .table table, body .ck-content .table td {
	border: dotted 1px #eee;
}
```

```css
/* page-with-editor-content.css */
body .ck-content .table table, body .ck-content .table td {
	border: none;
}
```

This will make the table visible when rendered in the editor even if it has no border set via the "Table properties" and "Table cell properties" balloons:

While ensuring that everything still works when the table is being styled via the balloons:

And that the border is not rendered at all if it was not set via the balloon when the content outputted from the editor is rendered on a target page (that uses `page-with-editor-content.css`).

## Should we do something with this?

It may be confusing for some developers why setting borders to none via the UI does not make the borders disappear completely. It'd be good to at least point in the documentation to this ticket.

However, perhaps we should do more?

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (table): Changed default table and table cell properties to match the content styles. Fixes problem with setting `border = none` on the table. Closes #6841. 

---

### Additional information
More context: #16810.
It's follow-up of this PR: #16002  
Fixes that have to be applied:

1.  https://github.com/ckeditor/ckeditor5/issues/16810#issuecomment-2258181045
2.  https://github.com/ckeditor/ckeditor5/pull/16002#issuecomment-2270562447

I decided to keep this in new PR due to change of ownership of this issue and keeping history of previous PR untouched.

#### Paste from Office test changes explanation

Aligned with actual data

```
<table class=MsoTableGrid border=1 cellspacing=0 cellpadding=0 width=597 style='border-collapse:collapse;mso-table-layout-alt:fixed;
border:none;
^^^^^^^^^
mso-border-alt:solid windowtext .5pt;mso-yfti-tbllook:1184;mso-padding-alt:
0cm 5.4pt 0cm 5.4pt'>
```

The first cell has a full border:

```
<td width="59%" style='width:59.0%;border:solid windowtext 1.0pt;mso-border-alt:
                                   ^^^^^^^^^^
solid windowtext .5pt;background:#CCCCCC;padding:0cm 5.4pt 0cm 5.4pt'>
```

While other cells have specific borders:

```
<td width="24%" style='width:24.0%;border:solid windowtext 1.0pt;
border-left:none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
^^^^^^^^^^^^^
background:#CCCCCC;padding:0cm 5.4pt 0cm 5.4pt'>
```

### CODE REVIEW

File: packages/ckeditor5-table/src/tablecellproperties/tablecellpropertiesediting.ts [hunk 1]
```diff
@@ -76,7 +76,11 @@ export default class TableCellPropertiesEditing extends Plugin {
 		const schema = editor.model.schema;
 		const conversion = editor.conversion;
 
-		editor.config.define( 'table.tableCellProperties.defaultProperties', {} );
+		editor.config.define( 'table.tableCellProperties.defaultProperties', {
```
Comment (niegowski) [inline packages/ckeditor5-table/src/tablecellproperties/tablecellpropertiesediting.ts:79]: These default values should be implemented in https://github.com/ckeditor/ckeditor5/blob/2b9341bb7932aaf3c1e55132abe940130d569cf9/packages/ckeditor5-table/src/utils/table-properties.ts#L100-L108

Comment (Mati365) [reply packages/ckeditor5-table/src/tablecellproperties/tablecellpropertiesediting.ts]: Yep, I was already doing this because I reopened old PR.

File: packages/ckeditor5-table/src/utils/table-properties.ts [hunk 2]
```diff
@@ -139,6 +139,8 @@ export function getNormalizedDefaultTableProperties(
 	options?: NormalizeTableDefaultPropertiesOptions
 ): NormalizedDefaultProperties {
 	return getNormalizedDefaultProperties( {
+		// It's workaround for the issue with missing support for border none in the table element.
```
Comment (Witoso) [inline packages/ckeditor5-table/src/utils/table-properties.ts:142]: I don't treat it as a workaround. It's a proper solution IMHO :)

**Review by Witoso — CHANGES_REQUESTED** [review]

**Review by Witoso — APPROVED** [review]

File: packages/ckeditor5-table/src/utils/table-properties.ts [hunk 3]
```diff
@@ -78,6 +78,17 @@ export interface NormalizedDefaultProperties {
 	horizontalAlignment?: string;
 }
 
+/**
+ * Options for the `getNormalizedDefaultProperties` function.
+ */
+type NormalizeTableDefaultPropertiesOptions = {
```
Comment (niegowski) [inline packages/ckeditor5-table/src/utils/table-properties.ts:84]: This type probably should be exported as the functions are using it and should have it documented.

Comment (niegowski) [reply packages/ckeditor5-table/src/utils/table-properties.ts:84]: Please also add docs for the fields as described in the original function parameter.

File: packages/ckeditor5-table/src/utils/table-properties.ts [hunk 4]
```diff
@@ -125,3 +130,39 @@ export function getNormalizedDefaultProperties(
...
+	options?: NormalizeTableDefaultPropertiesOptions
+): NormalizedDefaultProperties {
+	return getNormalizedDefaultProperties( {
+		// It adds support for border none in the table element, keep it in sync with the content styles
+		// See more: https://github.com/ckeditor/ckeditor5/issues/6841#issuecomment-1959195608
+		borderStyle: 'double',
+		borderColor: 'hsl(0, 0%, 70%)',
+		borderWidth: '1px',
+		width: '100%',
+		height: '100%',
```
Comment (niegowski) [inline packages/ckeditor5-table/src/utils/table-properties.ts:148]: I don't know if this is valid. By default, a table has no dimensions set.

File: packages/ckeditor5-table/tests/tablecellproperties/tablecellpropertiesediting.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ describe( 'table cell properties', () => {
 
 			expect( config ).to.be.an( 'object' );
 			expect( config ).to.have.property( 'defaultProperties' );
-			expect( config.defaultProperties ).to.deep.equal( {} );
+			expect( config.defaultProperties ).to.include( {} );
```
Comment (niegowski) [inline packages/ckeditor5-table/tests/tablecellproperties/tablecellpropertiesediting.js:49]: Why is this changed? What is the difference?

**Review by niegowski — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-table/tests/tablecellproperties/tablecellpropertiesediting.js (hunk 5 above)
Comment (Mati365) [reply packages/ckeditor5-table/tests/tablecellproperties/tablecellpropertiesediting.js:49]: oops, I reverted some code in this area and forgot to revert this comparison.

File: packages/ckeditor5-table/src/utils/table-properties.ts (hunk 4 above)
Comment (Mati365) [reply packages/ckeditor5-table/src/utils/table-properties.ts:148]: Yep looks like originally introduced [here](https://github.com/ckeditor/ckeditor5/issues/6841#issuecomment-1959195608). Works as expected without such styling. Thx

File: packages/ckeditor5-table/src/utils/table-properties.ts (hunk 3 above)
Comment (Mati365) [reply packages/ckeditor5-table/src/utils/table-properties.ts]: Docs adjusted

**Comment by Mati365** [comment]
@niegowski I applied suggestions

File: packages/ckeditor5-table/tests/utils/table-properties.js [hunk 6]
```diff
@@ -173,5 +177,31 @@ describe( 'table utils', () => {
...
+					borderStyle: 'double',
+					borderColor: 'hsl(0, 0%, 70%)',
+					borderWidth: '1px',
+					width: '',
+					height: ''
+				} );
+			} );
+		} );
+
+		describe( 'getNormalizedDefaultCellProperties', () => {
```
Comment (niegowski) [inline packages/ckeditor5-table/tests/utils/table-properties.js:194]:
```suggestion
		describe( 'getNormalizedDefaultCellProperties()', () => {
```

**Review by niegowski — APPROVED** [review]
